### ISSUE TITLE
VoiceOver gets stuck on the browser content "chrome" and doesn't enter the webpage

### ISSUE THREAD

**Issue by esbenam** [ckundo/auto-vo#6]
# Issue getting stuck on the page title

## What happens
When I run auto-vo it never moves the VO cursor into the actual webpage. VO gets stuck reading out the page title repeatedly since it stays on the element containing the actual webpage and doesn't enter the page itself.

## What should happen
VO should enter the webpage content and start navigating the page.

## Reproduction
I had to fix some script paths to test this, but once those were fixed it seems to happen for me every time I run `auto-vo --url https://smashingmagazine.com --limit 5` (limit has to be at least five to see the actual repetition happen).
 
## How can this be fixed?
I don't know if this is a configuration in my VO setup that needs to be changed (I don't remember if I am running the standard configuration or if I tweaked a few things over the years). When I use VO manually I can't just keep pressing `VO Right` and end up in the website content. I have to press `VO Shift Down` to actually enter the content.

### If this happens with a standard VO configuration
Is there a setting that needs to be documented aside from the "Allow VoiceOver to be controlled with AppleScript"? If so, what is it?

### If this doesn't happen with a standard VO configuration
Is this something that can be addressed in auto-vo, so multiple configurations can be handled automatically, or would the documentation have to include a set of "supported" settings?

## Did I just completely miss something
A very likely scenario - is this a "have you tried turning it off and on again"-thing on my end? What ever it is, I like the general idea of auto-vo, and look forward to taking it for a spin :)

**Comment by ckundo** [comment]
@esbenam this sounds related to the VoiceOver settings and how it navigates web content. could you open VoiceOver Utility, go to Web tab and take a screenshot?

**Comment by ckundo** [comment]
<img width="829" alt="Screen Shot 2021-06-24 at 3 53 05 PM" src="https://user-images.githubusercontent.com/108163/123323969-45a5a080-d504-11eb-80eb-ce59fe5ad2a8.png">
<img width="815" alt="Screen Shot 2021-06-24 at 3 18 19 PM" src="https://user-images.githubusercontent.com/108163/123323982-4b9b8180-d504-11eb-9321-594d9c0363af.png">

**Comment by esbenam** [comment]
Ahh, it seems to be the "Navigation" -> "Enable fast searching" feature. Is this enabled by default and I switched it off at some point? Either way, it might be a good idea to mention it in the readme, since the script relies on it being set (since there only is the `moveRight` command.)

**Comment by shaal** [comment]
I installed auto-vo by running `npm i -g auto-vo`
I made sure I have the same settings as in the screenshots above.

When I run `auto-vo --url https://smashingmagazine.com --limit 5` - a chrome page opens with smashingmagazine. "Welcome to MacOS, Link, VoiceOver off".
The chrome page closed automatically, and I'm back at the CLI with no output.

**Comment by SebDuf** [comment]
I have the same issue as @shaal, there was a prompt asking to allow Terminal to modify VoiceOver, but I could not find where in the settings to do this.

**Comment by ckundo** [comment]
@shaal @SebDuf these sound like separate issues. I'll take a look.

**Comment by ckundo** [comment]
@esbenam reopening since I think the `Enable Fast Searching` option is actually irrelevant (re #8). I toggled it off and restarted and the navigation works as expected.

**Comment by esbenam** [comment]
Okay. I will see if it was a random glitch when I am back on my computer monday, or if I can reproduce it consistently

**Comment by CarlyGerard** [comment]
I tried auto-vo out as well, using @ckundo's settings in the screenshots to be sure. I'm getting the following output in my command line when running `npx auto-vo --url https://www.wwu.edu/ --limit 10`: 

```
New Tab button
Search tabs pop up button
toolbar item palette
Western Washington University web content
Western Washington University web content
Western Washington University web content
Western Washington University web content
Western Washington University web content
Western Washington University web content
Western Washington University web content
```
I'm able to get it beyond the page title and hit the web content, but not actually go into web content.

**Comment by esbenam** [comment]
> @esbenam reopening since I think the `Enable Fast Searching` option is actually irrelevant (re #8). I toggled it off and restarted and the navigation works as expected.

Yeah, I can't reproduce it consistently. Sometimes auto-vo will stop and kill VO and Chromium without any output, and other times it will get stuck like this issue describes. It seems to consistently end up in the "kill without output" state after a couple of runs. Will update if I find anything new :)

**Comment by GregGuiver** [comment]
Would love to start using this, great idea. I am hitting the same issue as mentioned by @CarlyGerard though (I'm running Catalina for what it's worth) - I did wonder if the below settings under Web were correct?

![image](https://user-images.githubusercontent.com/80764909/124587978-2ab22500-de50-11eb-8d70-c1956ab0d2a0.png)

I can't seem to switch 'Automatically speak the web page' on using my mac but that's a whole other thing, if it's not important I'll stop chasing that issue down.

**Comment by ckundo** [comment]
When VoiceOver is on and the url changes, focus is given to the web content automatically. I think the issue here is that we were loading the url before VoiceOver had time to start. I updated the code to wait 1000 ms so that VoiceOver has some time to start before loading the page. I also added a change to wait for `domcontentloaded` before navigating the VO cursor (see #13 )

**Comment by ckundo** [comment]
just published v1.3.0 which switches to Webkit and uses the VoiceOver drill in mechanism to enter the web area correctly instead of waiting on timing.
